add event grid docs (#9034)
auth0/docs: articles/logs/streams/azure-event-grid.md

In your Azure portal:

1. Select Subscriptions
2. Select the subscription you’re using for Event Grid
3. On the left menu, under Settings, select Resource providers
4. Find Microsoft.EventGrid
5. Select Register
6. Refresh to make sure the status changes to Registered

### Set up an Auth0 event source

Part of the integration process is to set Auth0 up for use as an event source (this step happens on your Dashboard).

1. Log in to the [Auth0 Dashboard](${manage_url}).
2. Navigate to **Logs > Streams**.
3. Click **+ Create Stream**.
4. Select **Azure Event Grid** and enter a unique name for your new stream.
5. On the next screen, provide the following settings for your Event Grid stream:

| Setting | Description |
|---------|-------------|
| Name | A unique display name to distinguish this integration from other integrations. |
| Azure Subscription ID | The subscription ID is the unique alphanumeric string that identifies your Azure subscription. |
| Azure Region | The region where your Azure subscription is hosted. |
| Resource Group name | Name of the Azure resource group, which is entity to manage all Azure assets within one subscription. |

6. Click **Save**.

> Go to the [Azure Portal](https://portal.azure.com/#blade/HubsExtension/BrowseResource/resourceType/Microsoft.EventGrid%2FpartnerTopics) to complete the final steps of the integration.

Reviewer: This should be combined with **Activate your Auth0 Partner Topic in Azure**, but steps 1 and 2 of that section will take you to the same view as just clicking on this link directly.

> ### Set up an event handler

Reviewer: I don't think this step is really part of **Set up an Auth0 event source**. It's more part of **Subscribe to your Partner Topic**.

Go to your Azure subscription and spin up a service that is supported as an event handler (for a full list of all supported event handlers go to this [article](https://docs.microsoft.com/en-us/azure/event-grid/event-handlers)).

#### Activate your Auth0 Partner Topic in Azure

Activating the Auth0 topic in Azure allows events to flow from Auth0 to Azure.

1. Log in to the [Azure Portal](https://portal.azure.com/).
2. Search `Partner Topics` at the top and click `Event Grid Partner Topics` under services.
3. Click on the topic that matches the stream you created in your Auth0 Dashboard.
4. Confirm the `Source` field matches your Auth0 account.
5. Click Activate.

#### Subscribe to your Partner Topic

> You subscribe to an event grid topic to tell Event Grid which events to send to which event handler.

Reviewer: This line sounds kind of weird. Is this supposed to be imperative voice? Maybe just:
```suggestion
Subscribe to an Event Grid partner topic to tell Event Grid which events to send to your event handler.
```

> 1. On the Event Grid topic page, select + Event Subscription on the toolbar

Reviewer: ```suggestion
1. On the Event Grid partner topic Overview page, select **+ Event Subscription** on the toolbar.
```

---
title: Azure Event Grid Log Streams
description: Learn how to create an event-driven workflow using Azure Event Grid and send your tenant logs anywhere within the Azure ecosystem.
toc: false
topics:
 - logs
 - streams
 - event-streams
contentType: how-to
---

# Integrate Azure Event Grid with Auth0

> Azure’s Event Grid is a serverless event bus that acts as an intermediary allowing you to send data from Auth0 into the Azure ecosystem.

Reviewer 2: Maybe something like:

[Azure Event Grid](https://azure.microsoft.com/en-us/services/event-grid/) is a serverless event bus that lets you send event data from any source to any destination.

You can create event-driven workflows using to Event Grid send your Auth0 tenant logs to targets such as Azure Functions, Event Hubs, Sentinel, and Logic Apps.

You can create an event-driven workflow using Event Grid to send your Auth0 tenant logs to the targets of your choice (e.g., Azure Functions, Event Hubs, Sentinel and Logic Apps).

See the Auth0 event type codes for a [full list](https://auth0.com/docs/logs/references/log-event-type-codes) of the events that Auth0 supports.

## Send events from Auth0 to Azure Event Grid

> To send Auth0 events to Azure, you will need:

Reviewer 2: you'll need to:

1. Set up an event source (in this case, this is Auth0).
2. Set up an event handler, the app or service where the event will be sent.

For more information about these concepts, [see Concepts in Azure Event Grid](https://docs.microsoft.com/en-us/azure/event-grid/concepts)

### Set up an Auth0 event source

Part of the integration process is to set Auth0 up for use as an event source (this step happens on your Dashboard).
Log in to the Auth0 Dashboard.

1. Navigate to Logs > Streams.
2. Click + Create Stream.
3. Select Azure Event Grid and enter a unique name for your new stream.
4. Create the event source by providing your Azure Subscription ID, Azure Region and a Resource Group name. 
5. Click Save.

Go to the Azure Portal to complete the final steps of the integration.

## Set up an event handler

Go to your Azure subscription and spin up a service that is supported as an event handler (for a full list of all supported event handlers go to this [article](https://docs.microsoft.com/en-us/azure/event-grid/event-handlers)).

### Enable Event Grid resource provider

> If you haven’t previously used Event Grid, you will need to register the Event Grid resource provider. If you've used Event Grid before, skip to the next section.

Reviewer: This needs to come before the previous section (Set up an Auth0 event source), as we won't be able to create the partner topic if they don't have the event grid provider.

Reviewer: Maybe we should indicate this somewhere on the UI too. Or at least in the Tutorial tab.

Reviewer 3: Agreed. What do you think @camicano

Author: done

toc: false
topics:
 - logs
 - streams
 - event-streams
contentType: how-to
---

# Integrate Azure Event Grid with Auth0

Azure Event Grid is a serverless event bus that lets you send event data from any source to any destination.

You can create event-driven workflows using to Event Grid send your Auth0 tenant logs to targets such as Azure Functions, Event Hubs, Sentinel, and Logic Apps.

See the Auth0 event type codes for a [full list](https://auth0.com/docs/logs/references/log-event-type-codes) of the events that Auth0 supports.

## Send events from Auth0 to Azure Event Grid

To send Auth0 events to Azure, you'll need to:

1. Enable Event Grid resource provider.
2. Set up an event source (in this case, this is Auth0).
3. Set up an event handler, the app or service where the event will be sent.

For more information about these concepts, [see Concepts in Azure Event Grid](https://docs.microsoft.com/en-us/azure/event-grid/concepts)

### Enable Event Grid resource provider

If you haven’t previously used Event Grid, you will need to register the Event Grid resource provider. If you've used Event Grid before, skip to the next section.

In your Azure portal:

> 1. Select Subscriptions on the left menu

Reviewer: I'm not sure Subscriptions is guaranteed to be on the left nav, because I believe the items on there are dynamic based on what you've used recently. E.g. my left nav doesn't have it:

<img width="1642" alt="Screen Shot 2020-05-15 at 10 37 02" src="https://user-images.githubusercontent.com/48957/82079741-3369f980-9698-11ea-842f-6c45af81676c.png">

2. Select the subscription you’re using for Event Grid
3. On the left menu, under Settings, select Resource providers
4. Find Microsoft.EventGrid
5. Select Register
6. Refresh to make sure the status changes to Registered

### Set up an Auth0 event source

Part of the integration process is to set Auth0 up for use as an event source (this step happens on your Dashboard).

> Log in to the Auth0 Dashboard.

Reviewer 4: We could add a link to the auth0 manage link like we have on the other pages using `${manage_url}`

> 1. Navigate to Logs > Streams.

Reviewer 4: Adding this for consistency with other pages:

```suggestion
1. Navigate to **Logs > Streams**.
```

> 2. Click + Create Stream.

Reviewer 4: ```suggestion
2. Click **+ Create Stream**.
```

> 3. Select Azure Event Grid and enter a unique name for your new stream.

Reviewer 4: ```suggestion
3. Select **Azure Event Grid** and enter a unique name for your new stream.
```

> 4. Create the event source by providing your Azure Subscription ID, Azure Region and a Resource Group name. 

Reviewer 4: On the Datadog and HTTP pages, we use a table to describe each of these fields. Maybe we could do that here as well?

> 5. Click Save.

Reviewer 4: ```suggestion
5. Click **Save**.
```

> Go to the Azure Portal to complete the final steps of the integration.

Reviewer 4: >Go to the Azure Portal

Is there a link we could add in here?

Reviewer 3: @camicano this should work but please test https://portal.azure.com/?Microsoft_Azure_EventGrid_assettypeoptions=%7B%22PartnerTopic%22%3A%7B%22options%22%3A%22%22%7D%7D#blade/HubsExtension/BrowseResource/resourceType/Microsoft.EventGrid%2FpartnerTopics

Reviewer: That link has a feature flag in it. The go-live URL is https://portal.azure.com/#blade/HubsExtension/BrowseResource/resourceType/Microsoft.EventGrid%2FpartnerTopics.

Go to your Azure subscription and spin up a service that is supported as an event handler (for a full list of all supported event handlers go to this [article](https://docs.microsoft.com/en-us/azure/event-grid/event-handlers)).

#### Activate your Auth0 Partner Topic in Azure

Activating the Auth0 topic in Azure allows events to flow from Auth0 to Azure.

1. Log in to the [Azure Portal](https://portal.azure.com/).
2. Search `Partner Topics` at the top and click `Event Grid Partner Topics` under services.
3. Click on the topic that matches the stream you created in your Auth0 Dashboard.
4. Confirm the `Source` field matches your Auth0 account.
5. Click Activate.

#### Subscribe to your Partner Topic

You subscribe to an event grid topic to tell Event Grid which events to send to which event handler.

1. On the Event Grid topic page, select + Event Subscription on the toolbar
2. On the Create Event Subscription page:
    1. Enter a name for the event subscription.
    2. Select your desired Azure service or WebHook for the Endpoint type.
    3. Follow the instructions for the particular service.
    4. Back on the Create Event Subscription page, select Create.

> To send events to your topic, please follow the instructions on this article.

Author: missing link. I need to find it again

## Testing

At this point, your Event Grid workflow should be complete. 

### Verify the integration

To verify that the integration is working as expected:

1. Log in to the Auth0 Dashboard.

> 2. Navigate to Logs > Streams.

Reviewer 4: ```suggestion
2. Navigate to **Logs > Streams**.
```

3. Click on your Event Grid stream.

> 4. Once on the stream, click on the Health tab. The stream should be active and as long as you don't see any errors, the stream is working.

Reviewer 4: ```suggestion
4. Once on the stream, click on the **Health** tab. The stream should be active and as long as you don't see any errors, the stream is working.
```
